src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -410,19 +412,33 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 Case SyntaxKind.DollarSignDoubleQuoteToken
 
                     term = ParseInterpolatedStringExpression()
-
                 Case Else
-
                     If start.Kind = SyntaxKind.QuestionToken AndAlso CanStartConsequenceExpression(Me.PeekToken(1).Kind, qualified:=False) Then
                         ' This looks like ?. or ?! 
 
                         Dim qToken = DirectCast(start, PunctuationSyntax)
                         qToken = CheckFeatureAvailability(Feature.NullPropagatingOperator, qToken)
 
                         GetNextToken()
-                        term = SyntaxFactory.ConditionalAccessExpression(term, qToken, ParsePostFixExpression(RedimOrNewParent, term:=Nothing))
+                        term = SyntaxFactory.ConditionalAccessExpression(term, qToken, ParsePostFixExpression(redimOrNewParent, term:=Nothing))
                     Else
-                        If BailIfFirstTokenRejected Then
+                        If start.IsKeyword Then
+
+                            ' Is it in the form : [keyword] [qualifier] [identifier]
+                            ' If so treat the keyword as an identifier and continue to parse.
+                            Dim nextToken = PeekNextToken()
+                            If nextToken.Kind = SyntaxKind.DotToken OrElse nextToken.Kind = SyntaxKind.QuestionToken Then
+                                Dim keywordAsIdentifier = ParseSimpleNameExpressionAllowingKeywordAndTypeArguments()
+                                keywordAsIdentifier = ReportSyntaxError(keywordAsIdentifier, ERRID.ERR_InvalidUseOfKeyword)
+                                Dim _InvocationExpression_ = ParseExpressionCore(prevTerm:=keywordAsIdentifier)
+                                If _InvocationExpression_.Kind = SyntaxKind.InvocationExpression Then
+                                    Return _InvocationExpression_
+                                End If
+                            End If
+
+                        End If
```

Comment:
extra newline unnecessary. though you can potentially put a comment there saying it is intentional that you are falling through.